collect_app/src/test/java/org/odk/collect/android/formentry/RecordingHandlerTest.java

```diff
@@ -34,6 +39,7 @@ public class RecordingHandlerTest {
 
     private final FakeQuestionMediaManager questionMediaManager = new FakeQuestionMediaManager();
     private final FormController formController = mock(FormController.class);
```

Comment:
It feels like we really want `FormController` to be an interface we can create a fake for. Do you fancy pulling out that out and renaming our implementation to `JavaRosaFormController` as a follow-up PR? If we had a structure like that, you could have just created a fake that lets you pull out the value for an answer rather than needing Mockito. That'd definitely be useful in a bunch of places and I do remember feeling like there was too much mocking needed for this test when I was working on the feautre.

Reply:
Ok I've implemented such an interface. Please take a look and merge if everything is fine.